app/src/lib.rs

```diff
@@ -30,15 +30,42 @@ pub struct CommandRequest<C: DeserializeOwned + Serialize> {
     pub command: Option<C>,
 }
 
-/// An EventRequest may only be emitted by a server, of which there can be many, and
-/// only in relation to having received a [CommandRequest] from a client. Event requests
-/// take a type that provides their identifier; usually an enum. Event requests convey
+/// A temporal event is one that has its durability conveyed.
+pub trait TemporalEvent: Clone + DeserializeOwned + Serialize {}
+
+/// An event that has been logged, providing their identifier; usually an enum. These replies convey
 /// the offset they are associated with. If an offset overflows to zero then it is the
 /// server's responsibility to convey any important events that the client may need.
 /// It is the client's responsibility to clear state in relation to previous events when
 /// an offset less than or equal to the one it requested. An event request also conveys a
```

Comment:
Is there a missing word here? Perhaps something like `...previous events when **receiving** an offset...`

Reply:
I don't see there's anything missing...

Reply:
Sorry I highlighted the wrong bit. The whole sentence says:

`It is the client's responsibility to clear state in relation to previous events when
 /// an offset less than or equal to the one it requested.`
 
It seems like it needs a `receiving` in the middle or an `is received` at the end.

Reply:
Got it. Thanks.